src/firejail/seccomp.c

```diff
@@ -429,6 +429,35 @@ int seccomp_filter_mdwx(bool native) {
 	return 0;
 }
 
+// create namespaces filter
+int seccomp_filter_namespaces(bool native, const char *list) {
+	if (arg_debug)
+		printf("Build deny-new-namespaces filter\n");
+
+	const char *command, *filter;
+	if (native) {
+		command = "deny-new-namespaces";
+		filter = RUN_SECCOMP_NS;
+	} else {
+		command = "deny-new-namespaces.32";
+		filter = RUN_SECCOMP_NS_32;
+	}
+
+	// build the seccomp filter as a regular user
+	int rv = sbox_run(SBOX_USER | SBOX_CAPS_NONE | SBOX_SECCOMP, 4,
+					  PATH_FSECCOMP, command, filter, list);
```

Comment:
```suggestion
	// build the seccomp filter as a regular user
	int rv = sbox_run(SBOX_USER | SBOX_CAPS_NONE | SBOX_SECCOMP, 4,
	                  PATH_FSECCOMP, command, filter, list);
```

(Nitpick)

When aligning code based on the preceding line, please use tabs for indentation
and spaces for alignment so that the code looks aligned no matter how wide tabs
are configured to look (for example, using `ts=X` on vim).

On vim's default configuration (`ts=8`), the changed code looks unaligned:

```C
        // build the seccomp filter as a regular user
        int rv = sbox_run(SBOX_USER | SBOX_CAPS_NONE | SBOX_SECCOMP, 4,
                                          PATH_FSECCOMP, command, filter, list);
```

If setting `ts=4`, it looks aligned (so it looks like it assumes `ts=4`):

```C
    // build the seccomp filter as a regular user
    int rv = sbox_run(SBOX_USER | SBOX_CAPS_NONE | SBOX_SECCOMP, 4,
                      PATH_FSECCOMP, command, filter, list);
```

While, for example, all current invocations of `sbox_run()` in
`seccomp_filter_drop()` are aligned assuming `ts=8` (except for the last one
that is somehow aligned to `ts=7`):

<details>
<summary>src/firejail/seccomp.c</summary>
<p>

```C
// drop filter for seccomp option
int seccomp_filter_drop(bool native) {
        const char *filter, *postexec_filter;
// [...]
        if (cfg.seccomp_list_drop == NULL) {
                // default seccomp if error action is not changed
                if ((cfg.seccomp_list == NULL || cfg.seccomp_list[0] == '\0')
                    && arg_seccomp_error_action == DEFAULT_SECCOMP_ERROR_ACTION) {
// [...]
                }
                // default seccomp filter with additional drop list
                else { // cfg.seccomp_list != NULL
                        int rv;

                        if (arg_seccomp_block_secondary) {
                                if (arg_seccomp_error_action != DEFAULT_SECCOMP_ERROR_ACTION) {
                                        if (arg_debug)
                                                printf("Rebuild secondary block seccomp filter\n");
                                        rv = sbox_run(SBOX_USER | SBOX_CAPS_NONE | SBOX_SECCOMP, 4,
                                                      PATH_FSECCOMP, "secondary", "block", RUN_SECCOMP_BLOCK_SECONDARY);
                                        if (rv)
                                                exit(rv);
                                }
                                seccomp_filter_block_secondary();
                        } else {
#if defined(__x86_64__)
#if defined(__LP64__)
                                if (arg_seccomp_error_action != DEFAULT_SECCOMP_ERROR_ACTION) {
                                        if (arg_debug)
                                                printf("Rebuild 32 bit seccomp filter\n");
                                        rv = sbox_run(SBOX_USER | SBOX_CAPS_NONE | SBOX_SECCOMP, 4,
                                                      PATH_FSECCOMP, "secondary", "32", RUN_SECCOMP_32);
                                        if (rv)
                                                exit(rv);
                                }
                                seccomp_filter_32();
#endif
#endif
                        }
// [...]
                        // build the seccomp filter as a regular user
                        if (list && list[0])
                                if (arg_allow_debuggers)
                                        rv = sbox_run(SBOX_USER | SBOX_CAPS_NONE | SBOX_SECCOMP, 7,
                                                      PATH_FSECCOMP, command, "drop", filter, postexec_filter, list, "allow-debuggers");
                                else
                                        rv = sbox_run(SBOX_USER | SBOX_CAPS_NONE | SBOX_SECCOMP, 6,
                                                      PATH_FSECCOMP, command, "drop", filter, postexec_filter, list);
                        else
                                if (arg_allow_debuggers)
                                        rv = sbox_run(SBOX_USER | SBOX_CAPS_NONE | SBOX_SECCOMP, 4,
                                                      PATH_FSECCOMP, command, filter, "allow-debuggers");
                                else
                                        rv = sbox_run(SBOX_USER | SBOX_CAPS_NONE | SBOX_SECCOMP, 3,
                                                      PATH_FSECCOMP, command, filter);

                        if (rv)
                                exit(rv);

                        // optimize the new filter
                        rv = sbox_run(SBOX_USER | SBOX_CAPS_NONE | SBOX_SECCOMP, 2, PATH_FSEC_OPTIMIZE, filter);
                        if (rv)
                                exit(rv);
                }
        }

        // drop list without defaults - secondary filters are not installed
        // except when secondary architectures are explicitly blocked
        else { // cfg.seccomp_list_drop != NULL
// [...]
                // build the seccomp filter as a regular user
                int rv;
                if (arg_allow_debuggers)
                        rv = sbox_run(SBOX_USER | SBOX_CAPS_NONE | SBOX_SECCOMP, 6,
                                      PATH_FSECCOMP, command, filter, postexec_filter, list,  "allow-debuggers");
                else
                        rv = sbox_run(SBOX_USER | SBOX_CAPS_NONE | SBOX_SECCOMP, 5,
                                      PATH_FSECCOMP, command, filter, postexec_filter, list);

                if (rv)
                        exit(rv);

                // optimize the drop filter
                rv = sbox_run(SBOX_USER | SBOX_CAPS_NONE | SBOX_SECCOMP, 2, PATH_FSEC_OPTIMIZE, filter);
                if (rv)
                        exit(rv);
        }

        // load the filter
        if (seccomp_load(filter) == 0) {
                if (arg_debug)
                        printf("seccomp filter configured\n");
        }

        if (arg_debug && access(PATH_FSEC_PRINT, X_OK) == 0) {
                struct stat st;
                if (stat(postexec_filter, &st) != -1 && st.st_size != 0) {
                        printf("configuring postexec seccomp filter in %s\n", postexec_filter);
                        sbox_run(SBOX_USER | SBOX_CAPS_NONE | SBOX_SECCOMP, 2,
                                  PATH_FSEC_PRINT, postexec_filter);
                }
        }

        return 0;
}
```

</p>
</details>

**Note**: The excerpt from the last code block was taken from
src/firejail/seccomp.c at commit c7e4c8ed5 ("seccomp-log support in
firejail.config", 2022-06-18).

**Note2**: All tabs were converted to spaces on each code block to clarify how
it looks with different settings.

**Note3**: Tabs can be shown as ">" on vim by running `set list` (and hidden
again by running `set nolist`).  Other text editors may have similar toggles.

Reply:
Done.

Reply:
> If setting ts=4, it looks aligned (so it looks like it assumes ts=4)

Yeah that's it, I'm using `ts=4` for writing and usually check for inconsistencies with `ts=8` just before pushing changes. It's not really ideal and quite prone to indentation issues (as we see here).

Reply:
@smitsohu [on Jul 20](https://github.com/netblue30/firejail/pull/5259#discussion_r925431425):

> > If setting ts=4, it looks aligned (so it looks like it assumes ts=4)
> 
> Yeah that's it, I'm using `ts=4` for writing and usually check for
> inconsistencies with `ts=8` just before pushing changes. It's not really
> ideal and quite prone to indentation issues (as we see here).

I don't know about ideal (though I personally do find C code easier to visually
parse with `ts=8`), but using only spaces for alignment should generally
prevent alignment issues with any `ts=` value.

**Note**: Unlike in the first code block below, both calls currently look
aligned on the actual pull request diff (at commit d4a8354e8), but that's
because somehow GitHub manages to display tabs with `ts=9` on changed lines,
while unchanged lines are displayed with `ts=8`.

I see that with the force-push, one extra tab is used now to indicate line
continuation:

```diff
         // build the seccomp filter as a regular user
         int rv = sbox_run(SBOX_USER | SBOX_CAPS_NONE | SBOX_SECCOMP, 3,
-                 PATH_FSECCOMP, command, filter);
+                PATH_FSECCOMP, command, filter);

 // [...]

+        // build the seccomp filter as a regular user
+        int rv = sbox_run(SBOX_USER | SBOX_CAPS_NONE | SBOX_SECCOMP, 4,
+                PATH_FSECCOMP, command, filter, list);
```

I might use that myself depending on the occasion, but I'd note that at least
on this file, the `sbox_run` calls above would look different from most of the
other ones:

<details>
<summary>All sbox_run() calls in src/firejail/seccomp.c</summary>
<p>

```console
$ git grep -F -A 1 -h sbox_run c7e4c8ed59 -- src/firejail/seccomp.c |
  sed -e 's/^--//' -e '/if (/d' -e 's/\t/        /g'
                sbox_run(SBOX_USER | SBOX_CAPS_NONE | SBOX_SECCOMP, 2,
                        PATH_FSEC_PRINT, fname);

                                        rv = sbox_run(SBOX_USER | SBOX_CAPS_NONE | SBOX_SECCOMP, 4,
                                                      PATH_FSECCOMP, "secondary", "block", RUN_SECCOMP_BLOCK_SECONDARY);

                                        rv = sbox_run(SBOX_USER | SBOX_CAPS_NONE | SBOX_SECCOMP, 4,
                                                      PATH_FSECCOMP, "secondary", "32", RUN_SECCOMP_32);

                                        rv = sbox_run(SBOX_USER | SBOX_CAPS_NONE | SBOX_SECCOMP, 7,
                                                      PATH_FSECCOMP, command, "drop", filter, postexec_filter, list, "allow-debuggers");

                                        rv = sbox_run(SBOX_USER | SBOX_CAPS_NONE | SBOX_SECCOMP, 6,
                                                      PATH_FSECCOMP, command, "drop", filter, postexec_filter, list);

                                        rv = sbox_run(SBOX_USER | SBOX_CAPS_NONE | SBOX_SECCOMP, 4,
                                                      PATH_FSECCOMP, command, filter, "allow-debuggers");

                                        rv = sbox_run(SBOX_USER | SBOX_CAPS_NONE | SBOX_SECCOMP, 3,
                                                      PATH_FSECCOMP, command, filter);

                        rv = sbox_run(SBOX_USER | SBOX_CAPS_NONE | SBOX_SECCOMP, 2, PATH_FSEC_OPTIMIZE, filter);

                        rv = sbox_run(SBOX_USER | SBOX_CAPS_NONE | SBOX_SECCOMP, 6,
                                      PATH_FSECCOMP, command, filter, postexec_filter, list,  "allow-debuggers");

                        rv = sbox_run(SBOX_USER | SBOX_CAPS_NONE | SBOX_SECCOMP, 5,
                                      PATH_FSECCOMP, command, filter, postexec_filter, list);

                rv = sbox_run(SBOX_USER | SBOX_CAPS_NONE | SBOX_SECCOMP, 2, PATH_FSEC_OPTIMIZE, filter);

                        sbox_run(SBOX_USER | SBOX_CAPS_NONE | SBOX_SECCOMP, 2,
                                  PATH_FSEC_PRINT, postexec_filter);

        int rv = sbox_run(SBOX_USER | SBOX_CAPS_NONE | SBOX_SECCOMP, 5,
                 PATH_FSECCOMP, "keep", filter, postexec_filter, list);

                        sbox_run(SBOX_USER | SBOX_CAPS_NONE | SBOX_SECCOMP, 2,
                                  PATH_FSEC_PRINT, postexec_filter);

        int rv = sbox_run(SBOX_USER | SBOX_CAPS_NONE | SBOX_SECCOMP, 3,
                 PATH_FSECCOMP, command, filter);
```

</details>
</p>

Which is why I suggested this (1 tab + 18 spaces):

```diff
        // build the seccomp filter as a regular user
        int rv = sbox_run(SBOX_USER | SBOX_CAPS_NONE | SBOX_SECCOMP, 4,
                          PATH_FSECCOMP, command, filter, list);
```

Alternatively, this would also look aligned (1 tab + 9 spaces):

```diff
        // build the seccomp filter as a regular user
        int rv = sbox_run(SBOX_USER | SBOX_CAPS_NONE | SBOX_SECCOMP, 4,
                 PATH_FSECCOMP, command, filter, list);
```